driver/src/main/scala/api/commands/writeresult.scala

```diff
@@ -32,8 +32,7 @@ sealed trait WriteResult {
 object WriteResult {
   def lastError(result: WriteResult): Option[LastError] = result match {
```

Comment:
I would argue that even for individual write the api could be improved, ie. it return a `Future[WriteResult]` and `WriteResult` can contains errors. But in reality if the future is successful then the `WriteResult`  does not contains errors, because in that case a failed Future is used.
This lead to hesitation or duplication: the user does not really know from the type if it need to inspect the  `WriteResult` or the `Future`'s exception when looking for standard mongodb error codes.

So I would prefer we either always use `WriteResult`/`MultiBulkWriteResult` in case of mongo errors,
OR we use failed futures but in this case the `WriteResult` should no contains any errors.